.github/workflows/release.yml

```diff
@@ -1,52 +1,127 @@
+---
 name: Release
-on:
+
+'on':
   push:
+    paths-ignore:
+      - '**/*.md'
     tags:
-      - "v[0-9]+.[0-9]+.[0-9]+"
+      - 'v[0-9]+.[0-9]+.[0-9]+'
     branches:
-      - build-*
+      - 'build-*'
   workflow_dispatch:
 
 jobs:
   release:
     strategy:
       fail-fast: false
       matrix:
-        platform: [macos-latest, ubuntu-latest] #, windows-latest]
+        platform: [
+          macos-latest,
+          ubuntu-latest,
+        ]
+        # windows-latest,
+
     runs-on: ${{ matrix.platform }}
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
       - name: Node.js setup
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: 16
+          cache: 'yarn'
+
+      - name: Rust setup (native)
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
 
-      - name: Rust setup
+      - name: Rust setup (macOS/aarch64)
+        if: startsWith(runner.os,'macOS')
         uses: actions-rs/toolchain@v1
         with:
           toolchain: stable
+          target: aarch64-apple-darwin
+          override: true
 
-      - name: Install dependencies (ubuntu only)
-        if: matrix.platform == 'ubuntu-latest'
+      - name: Cache cargo files and outputs
+        uses: Swatinem/rust-cache@v2
+
+      - name: Install dependencies (Linux)
+        if: startsWith(runner.os,'Linux')
         run: |
           sudo apt-get update
-          sudo apt-get install -y libgtk-3-dev webkit2gtk-4.0 libappindicator3-dev librsvg2-dev patchelf
+          sudo apt-get install -y \
+            libgtk-3-dev \
+            webkit2gtk-4.0 \
+            libappindicator3-dev \
+            librsvg2-dev \
+            patchelf
+
+      - name: Install dependencies (macOS)
+        if: startsWith(runner.os,'macOS')
+        run: |
+          brew install openssl
+          echo "AARCH64_APPLE_DARWIN_OPENSSL_INCLUDE_DIR=/usr/local/opt/openssl/include" >> $GITHUB_ENV
+          echo "AARCH64_APPLE_DARWIN_OPENSSL_LIB_DIR=/usr/local/opt/openssl/lib" >> $GITHUB_ENV
+
       - name: Install app dependencies and build web
         run: |
           yarn
           cd ./gui-react
           yarn
 
       - name: Build the app
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
         uses: tauri-apps/tauri-action@v0
-
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
           tagName: v__VERSION__ # tauri-action replaces \_\_VERSION\_\_ with the app version
           releaseName: 'v__VERSION__'
           releaseBody: 'See the assets to download this version and install.'
           releaseDraft: true
-          prerelease: false
\ No newline at end of file
+          prerelease: false
+
+      - name: Build the app (Linux/testing)
```

Comment:
What's with the `(.../testing)` naming?

Reply:
It's the testing I was doing. I don't believe these bits are going to be run for the tagged releases. Or it would be run for everything, vs the releases which needs a tag starting with ```v```. Can't do releases unless it's tagged.

I was also thinking of doing an ARM64 Linux build, but had issues, request was really for Apple M1 build.